cmake/FindnetCDF.cmake

```diff
@@ -0,0 +1,158 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
```

Comment:
As far as I understand we need FindPackage.cmake only if Package does not provide a cmake file for us. However, netcdf does provide a CMakeLists.txt (as of version 4.3; see https://github.com/Unidata/netcdf-c) Is there a reason not to use that one?

Reply:
I think I was originally using a plain `find_package`, and @ZedThree recommended I include this. Beyond making nicer targets to link against, I've honestly forgotten what the rationale was -- I know it was something to do with `nc-config`, but I'm not sure about its significance. @ZedThree please could you explain it again?

Reply:
Yes, netCDF can be built with either autotools (`./configure`) or CMake, and commonly (or at least it used to be the case) HPC systems and OS package managers build it with autotools -- meaning the necessary cmake config files aren't present. It's not the `CMakeLists.txt` that's important, it's `netCDF-config.cmake`, which is generated by the configure process.

This module tries to find the cmake config file if present, and if not, falls back to using the `nc-config` script. I've found this module to be robust across many different systems.

Reply:
Could you add this as a comment into `FindnetCDF.cmake` then I will not forget it for future reference